lib/src/spawner.rs

```diff
@@ -110,14 +114,12 @@ impl Spawner {
             builder.insert(ConstantVelocity::new(facing.to_vec3() * speed));
         }
 
-        // Housekeeping
-        let new_hitbox = builder.id();
         if descriptor.attached_to_player {
             commands.entity(parent).push_children(&[new_hitbox]);
         }
-        self.spawned.insert(id, new_hitbox);
+        self.spawned.insert(new_hitbox.id(), new_hitbox);
         self.despawn_requests.push(DespawnRequest {
-            id,
+            id: new_hitbox.id(),
```

Comment:
Käytät tota id:tä useasti. Voisit tallentaa sen muuttujaan.

Reply:
Näemmä käytänkin. Siivoilen sen siitä

Reply:
Come to think of it, tarvitseeko toi OnHit juttu mitään ID:tä?